nav2_controller/src/nav2_controller.cpp

```diff
@@ -305,22 +306,70 @@ void ControllerServer::setPlannerPath(const nav_msgs::msg::Path & path)
     end_pose.pose.position.x, end_pose.pose.position.y);
 }
 
+double getDistance(
+  const geometry_msgs::msg::PoseStamped & pose_a,
+  const geometry_msgs::msg::PoseStamped & pose_b)
+{
+  double x_diff = fabs(pose_a.pose.position.x - pose_b.pose.position.x);
```

Comment:
You don't need the `fabs` since you're throwing them into `hypot`

Reply:
We also have a new utility in geometry_util in the nav2_util for this sort of thing, you should use it.